docs/src/tutorials/testing.md

```diff
@@ -1,243 +1,505 @@
+    gl1 = c.node()
+
+	# Start signer and connect to (l2)
+    s = c.signer().run_in_thread()
+    gl1.connect_peer(l2.info['id'], f'127.0.0.1:{l2.daemon.port}')
+
+	# Fund a channel (l2)----(gl).
+	# This results in the following network
+	# (l1)----(l2)----(gl)
+    l2.rpc.fundchannel(c.node_id.hex(), 'all')
+	# Generate a block to synchronize and proceed
+	# with the channel funding.
+    bitcoind.generate_block(1, wait_for_mempool=1)
+	# Wait for the channel to confirm.
+	wait_for(lambda:
+        gl1.list_peers().peers[0].channels[0].state == 'CHANNELD_NORMAL'
+    )
+
+    # Wait for all channels to appear in our view of the network. We 
+    # don't even have to wait for our channel to appear at the gossmap
+    # of l1: We can give a hint as soon as we know that our channel is 
+    # not a dead end. We wait for 4 entries in our gossmap as both 
+    # channels are bidirectional.
+    bitcoind.generate_block(5)
+    wait_for(
+        lambda: len([c for c in gl1.list_channels().channels]) == 4 
+    )
+   
+    # Create an invoice.
+    bolt11 = gl1.invoice(
+        amount_msat=clnpb.AmountOrAny(amount=clnpb.Amount(msat=100000)),
+	    description="test-desc",
+	    label ="test-label",
+    ).bolt11
+
+    # Pay invoice.
+    l1.rpc.pay(bolt11)
 ```
 
-To run this test you can use the above `Makefile` target:
+__Congratulations__! You have written your first test using the 
+`gl-testing` testing framework and a Greenlight node. Our test creates
+a small line graph network consisting of 3 lightning nodes with a 
+Greenlight node at the end. We created an invoice on the Greenlight
+node and payed for it with the first node in line.
+
+Let's check if our test passes!
+
+Remember that we need to call the tests from our docker shell
+```bash 
+make docker-shell
+```
 
+We can start our `pytest` test now. Remember the options `-v` and `-s`
+to print the logs to stdout instead of capturing them. 
 ```bash
-$ sudo make docker-image docker-shell  # (1)!
-$ make build-self check-self  # (2)!
+pytest -vs my-first-test.py
 ```
 
- 1. Enter the docker shell, building the image if it was not generated before.
- 2. In the docker shell, build `gl-client` and bindings, then execute
-    the self-tests
+After the test is finished you should see `passed` in the terminal.
+```bash
+========================================== 1 passed in 30.83s ===========================================
+```
 
-If you can drive your application from a python script through a UI
-testing framework or an API, you can just use the above test as a
-template and start writing your own tests.
+###You made it! 
 
-If you don't have a way to run your application through a scripted
-test you may want to start a network, and test against that
-manually. For an example of this see the next section.
+You can now buy the Blockstream [sticker pack][stickers] 
+and officially ennoble yourself with a "__I got Lightning working but 
+all I got was this sticker__" sticker on your laptop.
```

Comment:
Do we really make this guide an ad for the Blockstream Store? We should rather gift them the stickers if they made it through to here :-)